core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultMemoryHealthChecker.java

```diff
@@ -0,0 +1,44 @@
+package com.linecorp.armeria.server.healthcheck;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.util.List;
+
+public class DefaultMemoryHealthChecker implements HealthChecker {
+
+    private final List<MemoryPoolMXBean> memoryPoolMXBeans;
+    private final double targetMemoryHeapUsage;
+    private final double targetMemoryNonHeapUsage;
+    private final double targetMemoryTotalUsage;
+
+    public DefaultMemoryHealthChecker(final double targetMemoryHeapUsage, final double targetMemoryNonHeapUsage, final double targetMemoryTotalUsage) {
```

Comment:
Can we use ratio instead, ranging from 0.0 to 1.0? By using ratios, this check will be more reliable because it'll continue to work even after adjusting JVM options, etc.

Reply:
a5eccc9862241a33f4cafe1b40ea2373571c22dc
7403228a71684aa3234a5673d6e59e48545d98b7 
I changed it so that it checks the memory health using a ratio from 0.0 to 1.0. It may be a good idea to do an assertion that the value is between 0.0 and 1.0, or to add comments.

Reply:
Yeah, it'd be nice to add validation here.